controllers/liveupdate_controller.go

```diff
@@ -0,0 +1,653 @@
+			}
+			delete(selector, "version")
+			sourceClusterService.Spec.Selector = selector
+			return nil
+		})
+		if err != nil {
+			r.reqLogger.Error(err, "Failed to update service")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Remove version label from the pods
+	targetStatefulSet := &appsv1.StatefulSet{}
+	err := r.Get(r.ctx, types.NamespacedName{Namespace: r.infinispan.GetNamespace(), Name: r.targetName}, targetStatefulSet)
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "Failed to find the new statefulSet")
+		return reconcile.Result{}, err
+	}
+	podList, err := r.podList(targetStatefulSet)
+	if err != nil {
+		r.reqLogger.Error(err, "Error getting pods of the new statefulset")
+		return reconcile.Result{}, err
+	}
+
+	for _, pod := range podList.Items {
+		_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, &pod, func() error {
+			if pod.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(corev1.Resource(""), pod.Name)
+			}
+			labels := pod.ObjectMeta.Labels
+			delete(selector, "version")
+			pod.Labels = labels
+			return nil
+		})
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Error deleting pod '%s' version ", pod.Name)
+		}
+	}
+
+	// Store ping service name from the original cluster
+	oldPingName := ispn.GetPingServiceName()
+
+	// Change statefulSet reference in the Infinispan resource
+	_, err = controllerutil.CreateOrPatch(r.ctx, r, ispn, func() error {
+		if ispn.CreationTimestamp.IsZero() {
+			return errors.NewNotFound(schema.ParseGroupResource("infinispan.infinispan.org"), ispn.Name)
+		}
+		ispn.Status.StatefulSetName = r.targetName
+		return nil
+	})
+
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("error setting statefulSet '%s' reference to Infinispan", r.targetName)
+	}
+
+	// Delete ping service from the original cluster
+	pingService := &corev1.Service{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: ispn.Namespace, Name: oldPingName}, pingService)
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "source ping service  not found")
+		return reconcile.Result{}, err
+	}
+	err = r.Delete(context.TODO(), pingService)
+	if err != nil {
+		r.reqLogger.Error(err, "Error Deleting source ping service")
+		return reconcile.Result{}, err
+	}
+
+	// Delete old statefulSet
+	sourceStatefulSet := &appsv1.StatefulSet{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: ispn.Namespace, Name: ispn.Name}, sourceStatefulSet)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "source stateful set not found")
+		return reconcile.Result{}, err
+	}
+
+	err = r.Delete(r.ctx, sourceStatefulSet)
+	if err != nil {
+		r.reqLogger.Error(err, "Error Deleting statefulSet")
+		return reconcile.Result{}, err
+	}
+
+	// Remove any LiveUpdate status from the Infinispan CR
+	_, err = controllerutil.CreateOrPatch(r.ctx, r, ispn, func() error {
+		ispn.Status.LiveUpgradeStage = ""
```

Comment:
There's multiple updates to the `ispn.Status` in this method. I think we need to either return on the first call and requeue another reconcilliation, or add the `ispn.Status.StatefulSetName = r.targetName` here.